order-dev.js

```diff
@@ -0,0 +1,488 @@
+    if (chosen_proofs.includes(letter)) {
+      item.approved = true;
+    }
+    return item;
+  });
+
+  // look for proof_data to augment this - S3 link & timestamps
+  if (data.proof_data) {
+    var proofsData = {};
+    data.proof_data.split(',').forEach((proofString) => {
+      var proofData = proofString.split('|');
+      proofsData[proofData[0].trim()] = {
+        date: new Date(proofData[1]),
+        s3Link: proofData[2]
+      };
+    });
+    data.auto_proof_files = data.auto_proof_files.map((proof) => {
+      var letter = proof.filename.split('_')[1];
+      if (proofsData[letter]) {
+        proof.url = proofsData[letter].s3Link;
+        proof.date = proofsData[letter].date;
+      }
+      return proof;
+    });
+  }
+
+  // sort the proofs first
+  var sortedProofs = data.auto_proof_files.sort((a, b) => {
+    return (a.filename > b.filename) ? 1 : -1;
+  });
+
+  var result = `
+    <style>
+      .section {
+        position: relative !important;
+      }
+      .proof_overlay {
+        position: absolute;
+        background-color:rgba(0,0,0,0.2);
+        color: white;
+        border-radius: 7px;
+        pointer-events: none;
+      }
+      .proof_overlay.approved {
+        background-color: #1abf1a;
+      }
+      .proof_overlay p {
+        font-size: 0.8em;
+        line-height: 12px;
+        font-weight: bold;
+        white-space: nowrap;
+      }
+      .proof_overlay.top {
+        left: 11px;
+        top: 11px;
+      }
+      .proof_overlay.bottom {
+        right: 11px;
+        bottom: 18px;
+      }
+    </style>
+  `;
+  sortedProofs.forEach((proof) => {
+    result += Render.try('proof', proof);
+  });
+  return result;
+};
+
+Render.proof = (proof) => {
+  var letter = proof.filename.split('_')[1].toLowerCase();
+  var text = (proof.approved) ? `✔ ${letter.toUpperCase()}` : `${letter.toUpperCase()}`;
+  if (proof.date) text += ` • <span class="datetime">${proof.date}</span>`;
+  return `
+    <div class="section" proof="${letter}">
+      <div class="proof_overlay top ${(proof.approved) ? 'approved' : ''}" id="proof_${letter}">
+        <p>${text}</p>
+      </div>
+      <div class="proof_overlay bottom">
+        <p>• click to see full •</p>
+      </div>
+
+      <div class="row">
+
+        <figure>
+          <a href="${proof.url}">
+            <img src="${proof.url}" alt="Proof ${letter.toUpperCase()}"/>
+          </a>
+        </figure>
+      </div>
+    </div>
+  `;
+}
+function ticketItemString(data) {
+  var string = '';
+  string += `${data.custFirst} ${data.custLast},`;
+  string += `${data.email},`;
+  string += `${data.orderId_raw}`;
+
+  return string;
+}
+
+async function createReplyTicket() {
+  var replyText = $("textarea").val();
+  var reply = convertReply($("textarea").val());
+  API.clearCache('v2-getCustomerOrder-dev_' + window.location.search);
+  if(reply && reply.length > 9) {
+    $('div.customerReply').replaceWith(`
```

Comment:
I sense replicated logic here. This looks like the block from 
instead of replacing this whole section with a complete mock version of what was there...
why not just replace the button, with the spinner.
$('button.replyButton').replaceWith('<div class="spinner"></div>');